src/components/Homepage.tsx

```diff
@@ -11,12 +13,45 @@ import { PostType } from "../lib/types";
 
 const Homepage = () => {
   const navigateTo = useNavigate();
+  const [search, updateSearch] = useState("");
   const [query, updateQuery] = useState("");
+  const [advancedFiltersSelected, setAdvancedFiltersSelected] = useState(false);
 
   const submitHandler: React.FormEventHandler<HTMLFormElement> = (event) => {
     event.preventDefault();
 
-    navigateTo(`/facebook?q=${query}`);
+    if (advancedFiltersSelected) navigateTo(`/facebook?q=${query}`);
+    else navigateTo(`/facebook?q=${search}`);
+  };
+  const [name, updateName] = useState("");
+  const [unix, updateUnix] = useState("");
+  const [country, updateCountry] = useState("");
+  const [state, updateState] = useState("");
+  const [city, updateCity] = useState("");
+  const [year27Selected, setYear27Selected] = useState(false);
+  const [year26Selected, setYear26Selected] = useState(false);
+  const [year25Selected, setYear25Selected] = useState(false);
+  const [year24Selected, setYear24Selected] = useState(false);
```

Comment:
The years can be autogenerated. These states should be interpreted as first-years/sophomores/juniors/seniors, and then added by the base academic year.

The academic year can be calculated with these codes here. We may want to turn it into a util function for reuse.

https://github.com/WilliamsStudentsOnline/wso-react/blob/d82b8cee4f4442373459dbdc01f8d4ced20e845c/src/constants/constants.js#L25-L31

Reply:
Also, the state here could simply be a number, or an enum of four options (preferred) if you want.